pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/TextIndexHandler.java

```diff
@@ -146,7 +147,10 @@ private boolean shouldCreateTextIndex(ColumnMetadata columnMetadata) {
     if (columnMetadata != null) {
       // Fail fast upon unsupported operations.
       checkUnsupportedOperationsForTextIndex(columnMetadata);
-      return true;
+
+      // Create text index only for those segments that are created within recent time-window
```

Comment:
Looking at a sample segment metadata from ZK propertyStore:

```
{
  "id": "viewName__8__7050__20230601T0639Z",
  "simpleFields": {
    "segment.crc": "1173414070",
    "segment.creation.time": "1685601567635",
    "segment.download.url": "gs://root_dir/pinot-data/viewName/viewName__8__7050__20230601T0639Z",
    "segment.end.time": "1685604404829",
    "segment.flush.threshold.size": "2740205",
    "segment.index.version": "v3",
    "segment.realtime.endOffset": "18243008927",
    "segment.realtime.numReplicas": "2",
    "segment.realtime.startOffset": "18240268722",
    "segment.realtime.status": "DONE",
    "segment.start.time": "1685599558781",
    "segment.time.unit": "MILLISECONDS",
    "segment.total.docs": "2740205"
  },
  "mapFields": {},
  "listFields": {}
}
```
- `"segment.start.time": "1685599558781"`: `1 June 2023 06:05:58.781 GMT`
- `"segment.end.time": "1685604404829"`: `1 June 2023 07:26:44.829 GMT`
- `"segment.creation.time": "1685601567635"`: `1 June 2023 06:39:27.635 GMT``

As we can see, only `segment.creation.time` matches the date-time within segment name, i.e. `20230601T0639Z`